Default 2nd reporter type in dev mode to tap or use dev_mode_file_reporter conf

ventuno opened the pull request:
Fixes https://github.com/testem/testem/issues/1075.

See https://github.com/testem/testem/issues/1075#issuecomment-412674637 for a detailed description of the issue.

@johanneswuerbach I'm still not 100% sure this is the best solution. I like the idea that we are logging to a file, but assuming a `tap` reporter will always work fine seems wrong to me. What if the `report_file` is an `XML` file?

I think if we really want to have two reporters we might as well have a config property like `dev_mode_file_reporter` that the user can set to the preferred reporter type. Let me know what you think, otherwise this PR does just what we discussed on #1075.

johanneswuerbach commented:
I think in the long term creating multiple exports should be exposed in the config (the code should already support that), but I think that should be fine for now.

ventuno commented:
Thanks @johanneswuerbach, I'm not sure what you mean by "creating multiple exports should be exposed in the config", but if you think a new config `dev_mode_file_reporter` will be fine, I'll add that.

johanneswuerbach commented:
Yes, that also sounds fine for now.

> multiple exports should be exposed in the config

I was think of exposing the ability to specify an array of reports, which you could use like this:

```js
reporters = [{
   type: 'tap',
   output: 'stdout',
}, {
   type: 'junit',
   output: 'file',
   path: 'test_results.xml'
}]
```

This should be already supported within testem itself, but you can configure multiple reporters yet using the config / command line.

ventuno commented:
Thanks a lot @johanneswuerbach, I just added support for the `dev_mode_file_reporter` configuration option which lets the user define a custom `reporter` which will write to `report_file` in `dev` mode.
I also updated [config_file.md](https://github.com/testem/testem/blob/master/docs/config_file.md) with the documentation for the new option.

One last thing could be: do we really want to default to the `tap` reporter if none is specified or should we simply ignore it and maybe log a warning and invite the user to use the `dev_mode_file_reporter` option instead?

ventuno commented:
@johanneswuerbach any chance you could take a look at this?

johanneswuerbach commented:
Sorry, the warning sounds like a good idea.

ventuno commented:
Thanks @johanneswuerbach, I added the log as well. Would you mind taking reviewing this and see if this can be merged?

ventuno commented:
Thanks @johanneswuerbach any idea when the next release is going to happen?

johanneswuerbach commented:
Released as `v2.10.0`, thank you 👏 👍

ventuno commented:
Thanks @johanneswuerbach finally managed to test the latest version: works great!
